src/recensio/plone/behaviors/base_review.py

```diff
@@ -26,6 +27,9 @@ def generateDoi(context):
     Might not need to be called as a default factory, but just in the
     edit form
     """
+    if IBaseReview not in iterSchemata(context):
```

Comment:
I think it would be nicer, maybe also more performant‌ (which probably isn't a problem here) if you would use a marker interface.

You create a marker interface, register it for the behavior like so:

```
  <plone:behavior
      name="recensio.base_review"
      title="Base Review Behavior"
      factory=".base_review.BaseReview"
      provides=".base_review.IBaseReview"
      marker=".base_review.IBaseReviewMarker"
      for="plone.dexterity.interfaces.IDexterityContent"
      />
```

and then:
```suggestion
    if not IBaseReviewMarker.providedBy(context):
```